### PR TITLE
Fixes #32847 - job wizard description field with user data

### PR DESCRIPTION
**Pull request by MariaAga**
depends on: https://github.com/theforeman/foreman_remote_execution/pull/555 https://github.com/theforeman/foreman_remote_execution/pull/556

### CODE REVIEW

**Comment by MariaAga** [comment]
@laviro can you take a look? :)

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js [hunk 1]
```diff
@@ -13,18 +13,18 @@ export const DescriptionField = ({ inputs, value, setValue }) => {
         text: result[0],
       }));
       results.forEach(result => {
-        newDesc = newDesc.replace(
-          result.text,
-          // TODO: Replace with the value of the input from Target Hosts step
-          inputs.find(input => input.name === result.name)?.name || result.text
-        );
+        newDesc = newDesc.replace(result.text, inputValues[result.name] || '');
```
Comment (adamruzicka) [inline webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:16]: The old form left the formatting string untouched if it didn't match any input. If input command was ls and the template was 'Run %{command} %{foo}' then the rendered thing was `Run ls %{foo}`

**Review by adamruzicka — COMMENTED** [review]
One slight deviation from how it used to work, apart from it it works well

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js (hunk 1 above)
Comment (MariaAga) [reply webpack/JobWizard/steps/AdvancedFields/DescriptionField.js]: Changed and added tests for it

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js [hunk 2]
```diff
@@ -25,6 +34,10 @@ export const DescriptionField = ({ inputs, value, setValue }) => {
   const [generatedDesc, setGeneratedDesc] = useState(generateDesc());
   const [isPreview, setIsPreview] = useState(true);
 
+  useEffect(() => {
+    setGeneratedDesc(generateDesc());
```
Comment (laviro) [inline webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:38]: does `generateDesc` need to `useCallback` with `inputValues` as a dependency?

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js [hunk 3]
```diff
@@ -43,9 +56,20 @@ export const DescriptionField = ({ inputs, value, setValue }) => {
       }
     >
       {isPreview ? (
-        <TextInput id="description-preview" value={generatedDesc} isDisabled />
+        <Tooltip content={generatedDesc}>
+          <div>
+            {/* div wrapper so the tooltip will be shown in chrome */}
```
Comment (laviro) [inline webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:61]: is there an open issue for it in PF4?

File: webpack/JobWizard/steps/AdvancedFields/AdvancedFields.js [hunk 4]
```diff
@@ -22,10 +21,13 @@ import { DescriptionField } from './DescriptionField';
 import { WIZARD_TITLES } from '../../JobWizardConstants';
 import { WizardTitle } from '../form/WizardTitle';
 
-export const AdvancedFields = ({ advancedValues, setAdvancedValues }) => {
+export const AdvancedFields = ({
+  templateValues,
+  advancedValues,
+  setAdvancedValues,
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/JobWizard/steps/AdvancedFields/AdvancedFields.js:27]: Not sure if it's a good idea as `TemplateInputsFields` in this component only needs `advancedValues`

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js (hunk 2 above)
Comment (MariaAga) [reply webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:38]: as useeffect's only dependency is inputValues i'm not sure if there is a difference between using useeffect with `generateDesc` as a dependency and what we have now

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js (hunk 3 above)
Comment (MariaAga) [reply webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:61]: https://github.com/patternfly/patternfly-react/issues/1894

File: webpack/JobWizard/steps/AdvancedFields/__tests__/AdvancedFields.test.js [hunk 5]
```diff
@@ -155,5 +155,52 @@ describe('AdvancedFields', () => {
...
+      target: { value: descriptionValue },
+    });
+    await act(async () => {
+      fireEvent.click(screen.getByText('Preview job description'));
+    });
+    expect(
+      screen.getByLabelText('description preview', {
+        selector: 'input',
+      }).value
+    ).toBe('Run test command %{wrong command name}');
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/JobWizard/steps/AdvancedFields/__tests__/AdvancedFields.test.js:206]: Yup, added

File: webpack/JobWizard/steps/AdvancedFields/DescriptionField.js (hunk 2 above)
Comment (laviro) [reply webpack/JobWizard/steps/AdvancedFields/DescriptionField.js:38]: I believe you should add `generateDesc` to the `useEffect` dependencies,
and then it should `useCallback` with `[inputValues, inputs, value]` as dependencies

File: webpack/JobWizard/JobWizard.js [hunk 6]
```diff
@@ -113,7 +122,7 @@ export const JobWizard = () => {
               ...newValues,
             }));
           }}
-          jobTemplateID={jobTemplateID}
+          templateValues={templateValues}
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/JobWizard/JobWizard.js:125]: Is this the same question as the one above?

[the remark this answers is not in the document]
Comment (laviro) [reply webpack/JobWizard/JobWizard.js:125]: yes, but it can be resolved now :)

**Review by laviro — APPROVED** [review]
Thanks @MariaAga !

**Review by adamruzicka — APPROVED** [review]

**Pull request merged**